app/src/main/java/ch/epfl/sdp/blindly/LocationPermissionActivity.kt

```diff
@@ -0,0 +1,32 @@
+package ch.epfl.sdp.blindly
+
+import android.Manifest
+import android.content.Intent
+import android.content.pm.PackageManager
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.Button
+import androidx.core.app.ActivityCompat
+
+class LocationPermissionActivity : AppCompatActivity() {
+
+    private val FINE_LOCATION_PERMISSION_CODE = 2
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_location_permission)
+
+        val button: Button = findViewById(R.id.button)
+
+        button.setOnClickListener{
+            var permitted = ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED
+            while(!permitted) {
+                ActivityCompat.requestPermissions(this, arrayOf(Manifest.permission.ACCESS_FINE_LOCATION), FINE_LOCATION_PERMISSION_CODE)
+                permitted = ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED
+            }
```

Comment:
Is it really relevant to ask for permissions on a button click ?

Reply:
Yes because they have to read that the location is necessary to use the app